sycl/include/CL/sycl/detail/sycl_mem_obj_t.hpp

```diff
@@ -20,14 +20,17 @@
 #include <cstring>
 #include <type_traits>
 
+namespace pi {
```

Comment:
Please, define internal things inside cl::sycl::detail namespace.

Reply:
The PI library needs to be independent of SYCL, this code is forward declaring some PI members.

Reply:
So, then please, make sure this header is not directly or indirectly included when the user includes sycl.hpp.